### ISSUE TITLE
Sheen values look too dark

### ISSUE THREAD

**Issue by gkjohnson** [gkjohnson/three-gpu-pathtracer#363]
| PT | MV | DSPRT |
|---|---|---|
| ![](https://raw.githubusercontent.com/gkjohnson/three-gpu-pathtracer/screenshots/screenshots/golden/khronos-ToyCar.png) | ![](https://raw.githubusercontent.com/google/model-viewer/master/packages/render-fidelity-tools/test/goldens/khronos-ToyCar/model-viewer-golden.png) | |
| ![](https://raw.githubusercontent.com/gkjohnson/three-gpu-pathtracer/screenshots/screenshots/golden/khronos-GlamVelvetSofa.png) | ![](https://raw.githubusercontent.com/google/model-viewer/master/packages/render-fidelity-tools/test/goldens/khronos-GlamVelvetSofa/model-viewer-golden.png) | |
| ![](https://raw.githubusercontent.com/gkjohnson/three-gpu-pathtracer/screenshots/screenshots/golden/khronos-SheenCloth.png) | | ![](https://raw.githubusercontent.com/google/model-viewer/master/packages/render-fidelity-tools/test/goldens/khronos-SheenCloth/dspbr-pt-golden.png) |

**Comment by gkjohnson** [comment]
cc @richardassar - I know it's been a bit since this was added but any thoughts on the discrepancies compared to the Model Viewer models? The car and couch look particularly dark.

**Comment by richardassar** [comment]
I'll take a look, very busy at the moment with a few things but I will get to this when I have the oppotunity.
 
I do recall the kitchen towel model in Three GPU PathTracer looked more like what is currently in the DSPRT screenshot above, it had that purple sheen to it, so something might have changed since.

I'll let you know what I find.

**Comment by gkjohnson** [comment]
Of course! Whenever you have a moment or interest would be helpful if you can. I may take a deeper look when I get a chance, as well. I did a quick survey of how that cloth example changed over the releases since sheen was added (and notes on why the changes happened) and it looks like v0.0.6 -> v0.0.7 is the meaningful release. v0.0.7 fixed diffuse lighting contribution from being too dark which meant the sheen contribution did not stand out as much as it did in the previous release - perhaps the sheen value is also being scaled down somewhere? Or diffuse contribution needs to be scaled down where sheen contribution is high?

| version | screenshot | note |
|---|---|---|
| v0.0.6 | <img src="https://user-images.githubusercontent.com/734200/217519253-fd8ca6cc-b776-4c6c-94f0-b300303659b8.png" width="300px" /> | |
| v0.0.7 | <img src="https://user-images.githubusercontent.com/734200/217519660-dc682fd2-0cdb-4a7d-8c49-ee9272f80253.png" width="300px" /> | Base diffuse brightening correction likely from #255. |
| v0.0.8 | <img src="https://user-images.githubusercontent.com/734200/217520017-08fbb14b-004b-47b1-9560-afed35b3c45b.png" width="300px" /> | Roughness and metalness textures broken by #298 causing incorrect parameters and base contribution darkness. |
| v0.0.9 to v0.0.12 | <img src="https://user-images.githubusercontent.com/734200/217520308-3e0c845d-e1e1-4596-aa1d-19c091fb4610.png" width="300px" /> | Texture indexing fixed in https://github.com/gkjohnson/three-gpu-pathtracer/commit/2067567c346b2c250a40ee920119e4af39cacfa5.
| DSPRT | <img src="https://user-images.githubusercontent.com/734200/217521310-cf5e5ac7-9ab2-4578-908b-2fa8f70f528d.png" width="300px" /> |

**Comment by gkjohnson** [comment]
Two thoughts after taking a deeper look in #382: 
- It seems possible that sheen roughness should not have the "perceptual roughness" squaring applied (see [here](https://github.com/gkjohnson/three-gpu-pathtracer/blob/main/src/materials/PhysicalPathTracingMaterial.js#L805)).
- The [MeshPhysicalMaterial.sheen](https://threejs.org/docs/?q=physical#api/en/materials/MeshPhysicalMaterial.sheen) factor is not respected meaning you cannot turn sheen off at the moment.

Here's a comparison with and without the perceptual roughness applied on models where it makes a significant difference. The cloth model looks generally unchanged from this:

| sheenRoughness ** 2 | sheenRoughness | model-viewer |
|---|---|---|
| ![image](https://user-images.githubusercontent.com/734200/218293399-fb84941e-bdd8-48d9-a0a8-bd8e8f6a8e1c.png) | ![image](https://user-images.githubusercontent.com/734200/218293436-d3a8875d-9d8f-468d-9142-adaabe8ce94e.png) | ![image](https://user-images.githubusercontent.com/734200/218293460-ff091272-d610-45ba-b688-c851b192fdcf.png) |
| ![image](https://user-images.githubusercontent.com/734200/218293506-76adbcbf-9a97-4858-b410-3e5a610aba5f.png) | ![image](https://user-images.githubusercontent.com/734200/218293484-e6477bcb-e3ea-496f-bfc6-1221c64aa5d6.png) | ![image](https://user-images.githubusercontent.com/734200/218293474-f5e0dc24-3c7b-4555-a2a5-edc3c2994ab6.png) |

Note the weird hard-edged dark patches on the sofa are from a directional light casting hard shadows.

**Comment by gkjohnson** [comment]
> It seems possible that sheen roughness should not have the "perceptual roughness" squaring applied (see [here](https://github.com/gkjohnson/three-gpu-pathtracer/blob/main/src/materials/PhysicalPathTracingMaterial.js#L805)).

It looks like [three.js squares the sheen roughness](https://github.com/mrdoob/three.js/blob/8fab13b240999d157459f0caefbc5ee4357fd45d/src/renderers/shaders/ShaderChunk/bsdfs.glsl.js#L269) but I'm also seeing that the sheen roughness value is being squared inside the sheen functions already [here](https://github.com/gkjohnson/three-gpu-pathtracer/blob/4de53ebc08dffdb21dbb14beb5c9953b600978cc/src/shader/shaderSheenFunctions.js#L6-L7) and [here](https://github.com/gkjohnson/three-gpu-pathtracer/blob/4de53ebc08dffdb21dbb14beb5c9953b600978cc/src/shader/shaderSheenFunctions.js#L29-L30) - so we're currently squaring the roughness twice.

And here's the couch comparison without the directional light for a better comparison:

| Path Tracer | Three.js |
|---|---|
| ![image](https://user-images.githubusercontent.com/734200/218298430-a6786f21-2984-475c-a856-625538e23fa1.png) | ![image](https://user-images.githubusercontent.com/734200/218298440-f23f0336-2311-4017-9d69-1a91611f51ca.png) |
